docs/source/cuda/examples.rst

```diff
@@ -29,18 +32,23 @@ they may not be large enough to hold the entire inputs at once).
 
 It will be faster if we use a blocked algorithm to reduce accesses to the
 device memory.  CUDA provides a fast :ref:`shared memory <cuda-shared-memory>`
-for threads in a block to cooperately compute on a task.  The following
-implements a faster version of the square matrix multiplication using shared
-memory::
+for threads in a block to cooperatively compute on a task.  The following
+implements a faster version of the square matrix multiplication using shared memory:
+
+.. code-block:: python
+  :linenos:
 
     from numba import cuda, float32
 
     # Controls threads per block and shared memory usage.
     # The computation will be done on blocks of TPBxTPB elements.
-    TPB = 16
+    # TBP should not be larger than 32 in this example
+    TBP = 16
 
     @cuda.jit
     def fast_matmul(A, B, C):
+    """Based on corrected version by @RobertCrovella from https://stackoverflow.com/a/64198479/13697228
```

Comment:
We should try to keep code examples to within 80 characters so they'll pass flake8 checks. e.g.:

```suggestion
    """Reference: https://stackoverflow.com/a/64198479/13697228
    by @RobertCrovella
```